pyvo/mivot/viewer/mivot_viewer.py

```diff
@@ -0,0 +1,494 @@
+- extracts the annotation block from the VOTable
+- builds an XML view of the mapped model
+- updates the leaves of the XML view with the values read in the data rows
+- builds Python instances providing access to the mapped values by object attributes.
+
+The code below shows a typical use of `MivotViewer
+
+    .. code-block:: python
+    with MivotViewer(path_to_votable) as mivot_viewer:
+        print(f"mapped class id {mivot_instance.dmtype}")
+        print(f"space frame is  {mivot_instance.Coordinate_coordSys.spaceRefFrame.value}")
+
+        mivot_object = mivot_viewer.instance
+        while mivot_viewer.next():
+            print(f"latitude={mivot_object.latitude.value}")
+            print(f"longitude={mivot_object.longitude.value}")
+
+    See `tests/test_user_api.py`to get different examples of the API usage.
+"""
+import logging
+from copy import deepcopy
+from astropy import version
+from astropy.io.votable import parse
+from astropy.io.votable.tree import VOTableFile
+from pyvo.dal import DALResults
+from pyvo.mivot.utils.vocabulary import Ele, Att
+from pyvo.mivot.utils.vocabulary import Constant, NoMapping
+from pyvo.mivot.utils.exceptions import (MappingException,
+                                         ResourceNotFound,
+                                         MivotElementNotFound,
+                                         AstropyVersionException)
+from pyvo.mivot.utils.xml_utils import XmlUtils
+from pyvo.mivot.utils.xpath_utils import XPath
+from pyvo.mivot.seekers.annotation_seeker import AnnotationSeeker
+from pyvo.mivot.seekers.resource_seeker import ResourceSeeker
+from pyvo.mivot.seekers.table_iterator import TableIterator
+from pyvo.mivot.features.static_reference_resolver import StaticReferenceResolver
+from pyvo.mivot.version_checker import check_astropy_version
+from pyvo.mivot.viewer.mivot_instance import MivotInstance
+from pyvo.utils.prototype import prototype_feature
+from pyvo.mivot.utils.mivot_utils import MivotUtils
+from pyvo.mivot.viewer.xml_viewer import XMLViewer
+# Use defusedxml only if already present in order to avoid a new depency.
+try:
+    from defusedxml import ElementTree as etree
+except ImportError:
+    from xml.etree import ElementTree as etree
+
+
+@prototype_feature('MIVOT')
+class MivotViewer:
+    """
+    MivotViewer is a PyVO table wrapper aiming at providing
+    a model view on VOTable data read with usual tools.
+    """
+    def __init__(self, votable_path, tableref=None, resource_number=None):
+        """
+        Constructor of the MivotViewer class.
+        Parameters
+        ----------
+        votable_path : str, parsed VOTable or DALResults instance
+            VOTable that will be parsed with the parser of Astropy,
+            which extracts the annotation block.
+        tableref : str, optional
+            Used to identify the table to process. If not specified,
+            the first table is taken by default.
+        resource_number : int, optional
```

Comment:
Can this be automatically detected if not present in the first header? That would give more flexibility to users and data providers.

Reply:
If no resource_number is given, the first one is taken (see `_set_resource`). 
We didn't implement an automated search of the first annotated resource for 2 reasons:
1- Most of the VOTable have their data in the first resource
2- Such automation would complicate an already complex module, which is not necessarywhile we have no user feedback.

I think it is better to make later

Reply:
Hmm. The `query` module (https://github.com/astropy/pyvo/blob/555e5f3cba8aa673bbbd4a7d99cfb0a5df29eb29/pyvo/dal/query.py#L363) looks first for the resource with `results` type and after that defaults to 0 but that is not exposed to the user. This is meant to be used by someone that doesn't necessarily know (and it shouldn't) about the `VOTable` standard. 
I propose we remove the argument from here (it's always easier to add it later) and do what `query` does to guess the resource number. The resource number is an attribute of the class so someone can override it later.

Reply:
Resource_number parameter removed. 
MivotViewer automatically connect the first resource with @type=results